core/src/scene/scene.cpp

```diff
@@ -82,9 +83,25 @@ bool Scene::texture(const std::string& textureName, std::shared_ptr<Texture>& te
     return true;
 }
 
-void Scene::queueComponentUpdate(std::string componentPath, std::string value) {
-    std::vector<std::string> splitPath = splitString(componentPath, COMPONENT_PATH_DELIMITER);
-    m_updates.push_back({ splitPath, value });
+void Scene::queueUpdate(std::string path, std::string value) {
+    auto keys = splitString(path, COMPONENT_PATH_DELIMITER);
+    m_updates.push_back({ keys, value });
+}
+
+void Scene::addClientDataSource(std::shared_ptr<DataSource> _source) {
+    m_clientDataSources.push_back(_source);
+}
+
+void Scene::removeClientDataSource(DataSource& _source) {
+    auto it = std::remove_if(m_clientDataSources.begin(), m_clientDataSources.end(),
+        [&](auto& s) { return s.get() == &_source; });
+    m_clientDataSources.erase(it, m_clientDataSources.end());
```

Comment:
Yeah this is just how the "remove" algorithm works in the STL, "removing" means moving all the designated elements to the end of the collection and returning an iterator to the first element to be removed (http://en.cppreference.com/w/cpp/algorithm/remove). This trips me up all the time.